app.rb

```diff
@@ -0,0 +1,126 @@
+require './book'
+require './person'
+require './student'
+require './teacher'
+require './rental'
+
+class App
+  attr_accessor :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts "Book Store is currently empty" if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    puts "No person registered yet" if @people.empty?
+    @people.each do |person|
+      if person.is_a? Student
+        puts "[Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      elsif person.is_a? Teacher
+        puts "[Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print "Do you want to create a student (1) or teacher (2) [Input a number]: "
+    person = gets.chomp.to_i
+
+    case person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid input. Kindly type 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission [Y/N]? '
+    permission = gets.chomp.downcase
+
+    case permission
+    when 'y'
+      parent_permission = true
+    when 'n'
+      parent_permission = false
+    else
+      puts "Invalid input. Please enter 'Y' or 'N'"
+    end
+
+    student = Student.new(age, name, parent_permission: parent_permission)
+    @people << student
+
+    puts 'Student created successfully'
+  end
```

Comment:
- [ ] When I try to create a student through the UI, I get this error:

![image](https://user-images.githubusercontent.com/77106962/231870548-60f362ea-2ae4-448b-bf7c-fa919c56347d.png)

#### The way you are creating the new instance of the Student class is not correct. The only keyword argument available in the Student class during initialization is the classroom parameter and it has a default value. The correct way to create a new instance of the Student class would be:
` student = Student.new(age, name, parent_permission)`. Kindly implement this change. 😎 💻

Reply:
Hello, @Gambit142 Thank you for pointing this out. Although, if I create the Student in my console, it works perfectly fine, as shown in the picture, but I've still implemented the requested changes and I'll further look into it!
![library](https://user-images.githubusercontent.com/110582192/231884156-bb2863c9-5e8c-4b4c-9acd-b297b349009d.png)